### PR TITLE
Add libstemmer support to Groonga

### PR DESCRIPTION
**Pull request by nomoon**
- [X] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/master/CONTRIBUTING.md)?
- [X] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [X] Have you built your formula locally with `brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [X] Does your build pass `brew audit --strict <formula>` (after doing `brew install <formula>`)?

-----

This is necessary in order to be able to use the `TokenFilterStem` token filter.

### CODE REVIEW

**Comment by ilovezfs** [comment]
@BrewTestBot test this please

File: Formula/groonga.rb [hunk 1]
```diff
@@ -41,6 +43,11 @@ class Groonga < Formula
     sha256 "bc83d1e5e0f32d4b95e219cb940a7e3f61f0f743abd3bd47c2d436a34e503870"
   end
 
+  resource "stemmer" do
+    url "https://github.com/snowballstem/snowball.git",
+      :revision => "71936098048f915a797de990776564c924d40b6a"
```
Comment (MikeMcQuaid) [inline Formula/groonga.rb]: Any chance of addressing https://github.com/snowballstem/snowball/issues/7?

Comment (nomoon) [inline Formula/groonga.rb]: I wish. The official notes from the new Snowball page suggest that if a "stable" source distribution is required, one should use the old download available at http://snowball.tartarus.org/download.html, which is from 2014 and nonetheless still contains no numeric version number. There has been no update on @snowballstem/snowball#7 for over a year. I could ping them, however.

It appears that the [dovecot](https://github.com/Homebrew/homebrew-core/blob/master/Formula/dovecot.rb#L27) and [sphinx](https://github.com/Homebrew/homebrew-core/blob/master/Formula/sphinx.rb#L28) formulae have run into this same problem and also chose to pick an arbitrary commit for the resource.

Comment (nomoon) [inline Formula/groonga.rb]: Ubuntu, CentOS, etc. all have libstemmer packages which I believe are based on the 2014 source tar, which is certainly better than nothing, so I could alter the PR to use that if necessary.  groonga/groonga#430 suggests that the upstream project will eventually integrate libstemmer similarly to my implementation here.

**Comment by MikeMcQuaid** [comment]
Thanks for your contribution to Homebrew! Without people like you submitting PRs we couldn't run this project. You rock!

For future reference the preferred commit message format is `groonga: add libstemmer option.`. Please complete the issue template in future PRs where this format is detailed in the linked [Contributing guidelines](https://github.com/Homebrew/homebrew-core/blob/master/CONTRIBUTING.md). Thanks!

**Pull request closed without merging**